### PR TITLE
cli -addrinfo: drop torv2; torv3 becomes onion per GetNetworkName()

### PR DESCRIPTION
**Pull request by jonatack**
#22050 removed torv2 support from 22.0. For 23.0 and subsequent releases, we can probably remove torv2 from -addrinfo.

before
```
  "addresses_known": {
    "ipv4": 58305,
    "ipv6": 5138,
    "torv2": 0,
    "torv3": 5441,
    "i2p": 14,
    "total": 68898
  }
```
after
```
  "addresses_known": {
    "ipv4": 58305,
    "ipv6": 5138,
    "onion": 5441,
    "i2p": 14,
    "total": 68898
  }
```
Per the naming of `netbase.{h, cpp}::GetNetworkName()`, torv3 becomes onion, which is what is printed in the output of getpeerinfo, getnetworkinfo and getnodeaddresses.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -245,7 +245,7 @@ class BaseRequestHandler
 class AddrinfoRequestHandler : public BaseRequestHandler
 {
 private:
-    static constexpr std::array m_networks{"ipv4", "ipv6", "torv2", "torv3", "i2p"};
+    static constexpr std::array m_networks{"ipv4", "ipv6", "onion", "i2p"};
```
Comment (prayank23) [inline src/bitcoin-cli.cpp:248]: nit: maybe we can write torv3/v3 in bracket for few months

```suggestion
    static constexpr std::array m_networks{"ipv4", "ipv6", "onion(v3)", "i2p"};

**Comment by sipa** [comment]
Do you think "onion" is clearer than "tor"?

**Comment by jonatack** [comment]
I don't have a strong opinion, but in the absence of a needing to distinguish between tor/onion v2 and v3, using the default naming of GetNetworkName(), getnetworkinfo, getpeerinfo, and getnodeaddresses seemed to make sense and is the simplest to implement.

**Comment by sipa** [comment]
@jonatack Ah, being consistent with existing RPCs is a good reason.

**Comment by jonatack** [comment]
(Given the way "onion" is used by our Lightning Network colleagues as a more generic term, it may make sense to revert our language back to "tor" at some point.)

**Comment by laanwj** [comment]
> Do you think "onion" is clearer than "tor"?

As Tor allows connectivity to IPv4 and IPv6 as well, it's less specific. This was always the reasoning to use "onion" specifically when dealing with hidden services.

**Comment by jonatack** [comment]
> As Tor allows connectivity to IPv4 and IPv6 as well, it's less specific. This was always the reasoning to use "onion" specifically when dealing with hidden services. I don't think lightning also using the term is a good reason to change this.
> 
> (So concept ACK anyway)

Ah, thanks for the reminder why we did this.

File: src/bitcoin-cli.cpp (hunk 1 above)
Comment (jonatack) [reply src/bitcoin-cli.cpp:248]: That wouldn't work (but could be done by keeping and modifying the conditional that this patch removes). By the time 23.0 is released in 2022, my thought was that it should be widely known that only v3 onions are supported by Bitcoin Core and current releases of Tor, and "onion" is both simpler to implement and the same as the other RPCs.

**Comment by Zero-1729** [comment]
Concept ACK

**Comment by jonatack** [comment]
(Note to self, this needs an update to `doc/tor.md` and the release notes.)

**Comment by tryphe** [comment]
Concept ACK

"tor" seems maybe slightly more descriptive than "onion" but "onion" exists in the code more :man_shrugging:

**Comment by laanwj** [comment]
> "tor" seems maybe slightly more descriptive than "onion" but "onion" exists in the code more man_shrugging

The addresses are literally `.onion` addresses I don't understand why people suddenly don't find that descriptive enough anymore and want to change this around.

**Comment by jonatack** [comment]
Yes, I agree. Sorry for the mini-kerfuffle, I think "onion" is fine.

**Comment by tryphe** [comment]
> The addresses are literally `.onion` addresses I don't understand why people suddenly don't find that descriptive enough anymore and want to change this around. 

I wasn't saying I preferred one or the other, but rather I was neutral to the conversation above and didn't really mind either one.

**Comment by jonatack** [comment]
> (Note to self, this needs an update to `doc/tor.md` and the release notes.)

Done. This should be ready now.

**Review by Zero-1729 — APPROVED** [review]
tACK 49d503aefa74f11e5d93432987fa3775ed82c979 🧉

LGTM, patch works as described.

Updated Tor docs and release notes are clear and straightforward.

```sh
$ bitcoin-cli -addrinfo
{
  "addresses_known": {
    "ipv4": 39192,
    "ipv6": 3219,
    "onion": 11,
    "i2p": 0,
    "total": 42422
  }
}
```

**Review by klementtan — APPROVED** [review]
Code review and tested ACK 49d503aefa74f11e5d93432987fa3775ed82c979

```shell
{
  "addresses_known": {
    "ipv4": 83,
    "ipv6": 37,
    "onion": 30,
    "i2p": 0,
    "total": 150
  }
}
```

**Comment by practicalswift** [comment]
cr ACK 49d503aefa74f11e5d93432987fa3775ed82c979

**Pull request merged**